purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesOrchestrator.kt

```diff
@@ -195,9 +195,16 @@ internal class PurchasesOrchestrator(
     @SuppressWarnings("MagicNumber")
     private val lastSyncAttributesAndOfferingsRateLimiter = RateLimiter(5, 60.seconds)
 
+    @SuppressWarnings("MagicNumber")
+    private val preferredLocaleOverrideRateLimiter = RateLimiter(2, 60.seconds)
+
     var storefrontCountryCode: String? = null
         private set
 
+    @get:Synchronized
+    @set:Synchronized
+    var preferredUILocaleOverride: String? = initialConfiguration.preferredUILocaleOverride
```

Comment:
Hmm I'm not sure we should modify the `initialConfiguration` when modifying the locale... I think this should be a separate place to become the source of truth of the locale to be used IMO.